code/game/gamemodes/gang/gang.dm

```diff
@@ -0,0 +1,366 @@
+		readable_gang_names += "[G.name]"
+	var/finalized_gang_names = english_list(readable_gang_names)
+	priority_announce("Julio G coming to you live from Radio Los Spess! We've been hearing reports of gang activity on [station_name()], with the [finalized_gang_names] duking it out, looking for fresh territory and drugs to sling! Stay safe out there for the hour 'till the space cops get there, and keep it cool, yeah? Play music, not gunshots, I say. Peace out!", "Radio Los Spess", 'sound/voice/beepsky/radio.ogg')
+	sent_announcement = TRUE
+
+/datum/game_mode/gang/proc/five_minute_warning()
+	priority_announce("Julio G coming to you live from Radio Los Spess! The space cops are closing in on [station_name()] and will arrive in about 5 minutes! Better clear on out of there if you don't want to get hurt!", "Radio Los Spess", 'sound/voice/beepsky/radio.ogg')
+
+/datum/game_mode/gang/check_win()
+	var/alive_gangsters = 0
+	var/alive_cops = 0
+	for(var/datum/mind/gangbanger in gangbangers)
+		if(gangbanger.current)
+			if(!ishuman(gangbanger.current))
+				continue
+			var/mob/living/carbon/human/H = gangbanger.current
+			if(H.stat)
+				continue
+			alive_gangsters++
+	for(var/datum/mind/bacon in pigs)
+		if(bacon.current)
+			if(!ishuman(bacon.current)) // always returns false
+				continue
+			var/mob/living/carbon/human/H = bacon.current
+			if(H.stat)
+				continue
+			alive_cops++
+	if(alive_gangsters > alive_cops)
+		SSticker.mode_result = "win - gangs survived"
+		SSticker.news_report = GANG_OPERATING
+		return TRUE
+	else
+		SSticker.mode_result = "loss - police destroyed the gangs"
+		SSticker.news_report = GANG_DESTROYED
+		return FALSE
+
+/datum/game_mode/gang/process()
+	if(sent_announcement)
+		for(var/mob/M in GLOB.mob_list)
+			if(M.client)
+				if(M.hud_used)
+					var/datum/hud/H = M.hud_used
+					var/icon_state_to_use = "wanted_1"
+					if(lock_stars)
+						H.wanted_lvl.icon_state = current_stars
+					else
+						if(GLOB.joined_player_list.len > LOWPOP_FAMILIES_COUNT)
+							switch(GLOB.deaths_during_shift)
+								if(0 to TWO_STARS_HIGHPOP-1)
+									icon_state_to_use = "wanted_1"
+								if(TWO_STARS_HIGHPOP to THREE_STARS_HIGHPOP-1)
+									icon_state_to_use = "wanted_2"
+								if(THREE_STARS_HIGHPOP to FOUR_STARS_HIGHPOP-1)
+									icon_state_to_use = "wanted_3"
+								if(FOUR_STARS_HIGHPOP to FIVE_STARS_HIGHPOP-1)
+									icon_state_to_use = "wanted_4"
+								if(FIVE_STARS_HIGHPOP to INFINITY)
+									icon_state_to_use = "wanted_5"
+						else
+							switch(GLOB.deaths_during_shift)
+								if(0 to TWO_STARS_LOW-1)
+									icon_state_to_use = "wanted_1"
+								if(TWO_STARS_LOW to THREE_STARS_LOW-1)
+									icon_state_to_use = "wanted_2"
+								if(THREE_STARS_LOW to FOUR_STARS_LOW-1)
+									icon_state_to_use = "wanted_3"
+								if(FOUR_STARS_LOW to FIVE_STARS_LOW-1)
+									icon_state_to_use = "wanted_4"
+								if(FIVE_STARS_LOW to INFINITY)
+									icon_state_to_use = "wanted_5"
```

Comment:
If you decide to not make the screen object static, you should still migrate these switches out of the loop. No need to calculate the icon state for every single player.

Reply:
i tried this method and ran into very weird bugs with it not wanting to appear on some user's screens while still appearing on others, going to pass for now because that method is unstable

Reply:
You ran into issues with making it static or with migrating the switch out of the loop? Happy to believe the former but the second one seems unlikely.

Reply:
making it static